### ISSUE TITLE
The last update broke shortcut keys

### ISSUE THREAD

**Issue by sleek22** [mikecrittenden/shortkeys#290]
I was using F2/F3 to tab left and right. I was using F4 to close tabs to the left. I was also using F8 to run a script. When I updated to version 3.0.4, none of the shortcuts work anymore.

I uninstalled and installed the add-on again but no change so I reverted back to the earlier version which still works.

**Comment by mikecrittenden** [comment]
Interesting. I wonder if this is specifically related to the F keys. I'll take a look tomorrow.

**Comment by Nairo77** [comment]
Hello, I have a similar problem. I use shortcut key only to disable firefox rapid search shortcut (with the sign ' on the 4) and it doesn't work since the last update.

**Comment by mikecrittenden** [comment]
@sleek22 You on Firefox or Chrome? Windows or Mac? Can I see a screenshot of your settings?

I just tested some F keys and they worked for me. Example: 

![image](https://user-images.githubusercontent.com/32261/71600833-c0b01580-2b1e-11ea-9fd9-495d05ed720a.png)

**Comment by sleek22** [comment]
Hi, I'm on Firefox version 70.0.1 running on Windows 10. Below is a snapshot of my settings:

![pic](https://user-images.githubusercontent.com/4625685/71611698-66cc4180-2b58-11ea-9ca0-4848a4e88a94.png)

**Comment by mikecrittenden** [comment]
@sleek22 Maybe try using `F3` instead of `f3` - meaning use uppercase F instead of lowercase. I seem to remember that possibly being a fickle thing in the past with the Mousetrap library we're using.

@Nairo77 I'm not sure what "with the sign ' on the 4" means exactly, can you elaborate? Is that on a US keyboard?

**Comment by sleek22** [comment]
I changed the letters to capital and there was no change. When I restarted Firefox, it would work intermittently. For example, F3 will sometimes give next tab, at other times it opens up a search bar at the bottom of the browser page, which I believe is the standard behavior for the F3 command.

![pic 1](https://user-images.githubusercontent.com/4625685/71631889-961a9700-2bc8-11ea-8ba7-85b965510565.png)
![pic 2](https://user-images.githubusercontent.com/4625685/71631891-9b77e180-2bc8-11ea-8422-883bd8030306.png)

**Comment by mikecrittenden** [comment]
Oh interesting. In that case I don't think there's really much shortkeys
can do, if Firefox is sometimes hijacking the command before it can reach
Shortkeys. Whenever that happens I always just tell people to try to pick a
different keyboard shortcut that isn't already taken by the browser if
possible.

**Comment by sleek22** [comment]
The older version seems to be more robust as it has always worked flawlessly in that regard, at least for me. In any event, I'll just keep using the older version as it does everything I want it to do, and allows me to use single keys for shortcuts. (Using the F2/F3 combination for moving across tabs was something originally used by the Maxthon browser. This allows you to keep the right hand on the mouse and use the fingers of the left hand to move across tabs, etc.)

Also, thank you for making this wonderful add-on for us.

**Comment by mikecrittenden** [comment]
> The older version seems to be more robust as it has always worked flawlessly in that regard

Bummer. I am not sure what about the newer version could have changed that, except maybe that we upgraded https://github.com/ccampbell/mousetrap/releases to the latest version. I'll poke around and see if that could have done it.

**Comment by raydavis** [comment]
The update broke my shortcuts as well, all of which use "alt+DIGIT" as a hotkey. This is on Windows/Firefox.
![Shortkeys-screenshot](https://user-images.githubusercontent.com/98528/71633150-67e88780-2bc7-11ea-9f4b-b4d69ff881f5.png)

**Comment by mikecrittenden** [comment]
@raydavis are you able to get ANY shortcuts to work, with any combination? Even something simple like scroll up/down? I'm trying to narrow down the issue here.

**Comment by raydavis** [comment]
Understood, @mikecrittenden -- Since my existing shortcuts were all based on bookmarklets, I added an "alt+5" hotkey for "Scroll to bottom". After saving and reloading, that shortcut worked -- and the other, already defined, hotkeys revived as well. I wonder if the upgrade requires explicitly re-saving existing settings as well as requiring the extra permission....

**Comment by mikecrittenden** [comment]
Well that's interesting. I'm glad you're back up and running. But it sounds like re-saving didn't do squat for @sleek22. Oh well, I'll keep digging. Thanks for the info!

**Comment by ivanschou** [comment]
Not sure if it's ShortcutKeys or Firefox 70.  I'm running ShortcutKeys and recently updated everything (Firefox 70 and ShortcutKeys 3.0.4 on MacOS 10.14.6 and nothing works.  I created simple ctrl+j scroll down short cut and it does nothing.  I tried changing it to "Run Javascript" and using the script

console.log("Hello, World!"); 

and it doesn't look like it ever triggers.

I also have TamperMonkey, uBlock Origins, and Privacy Pass installed.  I tried disabling everything except for ShortcutKeys and still doesn't work.  I tried downgrading to 2.2.7 and that doesn't seem to work either.  Is it possible Firefox 70 did something?

**Comment by zoidbergthepopularone** [comment]
I have the same issue: After the last update, shortcuts "1" and "2" (just the numbers, no ALT or anything) for moving to the Previous and Next tabs stopped working on Firefox 71. I suspect it may be related to the new "This action is supported from Chrome's keyboard shortcuts settings", because even that menu won't allow me to enter a shortcut without modifiers. It would be nice if the previous version's solution was provided as a fallback.

**Comment by Nairo77** [comment]
@mikecrittenden Sorry, I wasn't at home. 
I'm on linux (debian - don't know wich one :/) with firefox (an old version 60.7.0) and french keybord (the "4" above the letters, not on the numerik keypad)
This morning, it works without that I have changed anything.

**Comment by mikecrittenden** [comment]
The whole "suddenly it started working again" thing seems to be a common theme here. I'm not really sure what to make of this issue. 

@zoidbergthepopularone FWIW, you should still be totally free to ONLY use the Shortkeys options page and completely ignore FF's built in Keyboard Shortcuts UI, just like before. The only change was to "turn on" the FF built in Keyboard Shortcuts UI for Shortkeys (it's been enabled for Chrome for a long time) but nothing was done to cripple or disable the old way of doing things.

**Comment by zoidbergthepopularone** [comment]
Unfortunately, with 3.xx, I can't use these shortcuts. Returned to version 2.2.7, disabled and enabled the extension and everything works fine again.

**Comment by franfernan** [comment]
I've found a fix that works for myself. I had the same problem. None of my shortcuts would work. I had to save or add a new shortcut, then everything started working again.

**Comment by ivanschou** [comment]
I've tried adding new shortcuts and modifying existing ones and also changing the access for individual shortcuts.  Doesn't work for me.  I'm looking at the "Debug Add-Ons" page, and I see this warning:

Warning details

Reading manifest: Warning processing options_page: An unexpected property was found in the WebExtension manifest.

I don't know if that means anything

**Comment by mikecrittenden** [comment]
Anyone have any new info? Or is anyone still having problems with the latest version?

**Comment by zoidbergthepopularone** [comment]
Sorry, cannot tell - it seems that my downgrade to 2.x didn't go entirely well, so now I am not notified of any new versions. And I am a bit worried about trying when 2.x works just fine for me.

**Comment by JohnDohe** [comment]
Hello,

I cannot install v 3.0.7

I got an email advising of the newest version which 
I downloaded for FF 72.0.1 (Win10Pro).

I extracted the zip file and tried to install.  Could not.
Th extracted contents do not have an xpi file FF is 
looking for.

I then uninstalled v 3.0.4 and tried to re-install from 
the FF extensions site,
https://addons.mozilla.org/en-US/firefox/addon/shortkeys/?src=search

All this did was install v 3.0.4 again (and I lost all my 
customized shortcuts).

What to do?

How to get the latest version 3.0.7 installed?

Thank you.
